### PR TITLE
docs: Add Web UI documentation on repo structure

### PR DESCRIPTION
**Pull request by adchia**
**What this PR does / why we need it**:
This adds a lot more documentation on the Feast Web UI documentation and the release process. Also updates the feast ui bundled UI to always use the latest npm package

**Which issue(s) this PR fixes**:

Fixes #

### CODE REVIEW

**Comment by adchia** [comment]
cc @AlexEijssen

File: ui/CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+- to be part of the @feast-dev team in NPM. Ask `#feast-development` on http://slack.feast.dev to add you if necessary. 
+- to [login to your NPM account on the command line](https://docs.npmjs.com/cli/v8/commands/npm-adduser).
+
+### Steps for Publishing
+
+1. Make sure tests are passing. Run tests with `yarn test` in the ui directory.
+2. Bump the version number in `package.json` as appropriate.
+3. Package the modules for distributions. Run the library build script with `yarn build:lib`. We use [Rollup](https://rollupjs.org/) for building the module, and the configs are in the `rollup.config.js` file.
+4. Publish the package to NPM. Run `npm publish`
+5. [Check NPM to see that the package was properly publish](https://www.npmjs.com/package/@feast-dev/feast-ui).
```
Comment (felixwang9817) [inline ui/CONTRIBUTING.md:99]: properly published

File: ui/CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+- [Release process](#release-process)
+  - [(Advanced) Manually publishing the Feast Package to NPM](#advanced-manually-publishing-the-feast-package-to-npm)
+    - [Requirements](#requirements)
+    - [Steps for Publishing](#steps-for-publishing)
+
+# General contributor notes
+In this doc, we describe how to contribute both to the Feast Web UI NPM package as well as the embedded Feast UI in the Python SDK (i.e. what's run when you run `feast ui`)
+
+## `feast ui` command
+You can see the logic in [../sdk/python/ui](../sdk/python/feast/ui/). This instance is loaded in [../sdk/python/ui_server.py](../sdk/python/feast/ui_server.py).
```
Comment (felixwang9817) [inline ui/CONTRIBUTING.md:20]: I think you mean `../sdk/python/feast/ui` and `../sdk/python/feast/ui_server.py`

Comment (adchia) [reply ui/CONTRIBUTING.md]: done

File: ui/CONTRIBUTING.md (hunk 1 above)
Comment (adchia) [reply ui/CONTRIBUTING.md]: done

**Review by felixwang9817 — APPROVED** [review]
/lgtm

**Pull request merged**